api/marc.py

```diff
@@ -41,21 +41,32 @@ def annotate_work_record(self, work, active_license_pool, edition,
             if include_genres:
                 self.add_simplified_genres(record, work)
 
-        self.add_web_client_urls(record, self.library, identifier)
+        self.add_web_client_urls(record, self.library, identifier, integration)
 
-    @classmethod
-    def add_web_client_urls(cls, record, library, identifier):
+    def add_web_client_urls(self, record, library, identifier, integration=None):
         _db = Session.object_session(library)
+        settings = []
+
+        if integration:
+            marc_setting = self.value(MARCExporter.WEB_CLIENT_URL, integration)
+            if marc_setting:
+                settings.append(marc_setting)
+
         from api.registry import Registration
-        settings = _db.query(ConfigurationSetting).filter(ConfigurationSetting.key==Registration.LIBRARY_REGISTRATION_WEB_CLIENT, ConfigurationSetting.library_id==library.id)
+        settings += [s.value for s in _db.query(
+            ConfigurationSetting
+        ).filter(
+            ConfigurationSetting.key==Registration.LIBRARY_REGISTRATION_WEB_CLIENT,
+            ConfigurationSetting.library_id==library.id
+        ) if s.value]
+
```

Comment:
Is this a different way of writing the same code as before? It looks like it to me but I want to make sure.

Reply:
I guess it lets you filter out empty values earlier.

Reply:
Yeah, it's pretty much the same, just reorganized.